spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -65,7 +65,8 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
         _: DateType | _: BooleanType | _: NullType =>
       true
     case dt if isTimestampNTZType(dt) => true
-    case s: StructType if allowStruct => s.fields.map(_.dataType).forall(supportedDataType(_))
+    case s: StructType if allowStruct =>
+      s.fields.map(_.dataType).forall(supportedDataType(_, allowStruct))
```

Comment:
Good catch, I initially had `allowStruct` default to true which is why I didn't pass it on here, but missed this when I changed the default